eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -98,6 +99,25 @@ case class NodeParams(nodeKeyManager: NodeKeyManager,
   def initFeaturesFor(nodeId: PublicKey): Features[InitFeature] = overrideInitFeatures.getOrElse(nodeId, features).initFeatures()
 }
 
+case class PaymentFinalExpiryConf(min: CltvExpiryDelta, max: CltvExpiryDelta) {
+  require(min.toInt >= 0, "cltv-expiry-delta must be positive")
+  require(min <= max, "maximum cltv-expiry-delta cannot be smaller than minimum-cltv-expiry-delta")
+
+  /**
+   * When sending a payment, if the cltv expiry used for the final node is very close to the current block height, it
+   * lets intermediate nodes figure out their position in the route. To protect against this, a random delta is added
+   * to the current block height, which makes it look like there are more hops after the final node.
+   */
+  def computeFinalExpiry(currentBlockHeight: BlockHeight, minFinalExpiryDelta: CltvExpiryDelta): CltvExpiry = {
+    val additionalDelta = if (min < max) {
+      min + (randomLong() % (max - min).toInt).toInt.abs
```

Comment:
```suggestion
      min + (randomLong() % (max - min + 1).toInt).toInt.abs
```
You should add 1 so that the max is reachable.

Reply:
Good idea, that makes more sense. Done in https://github.com/ACINQ/eclair/pull/2469/commits/292a069e96fe65cf0bcefdc8495a0b418c069354